### PR TITLE
Setup additional public and private subnet

### PR DESCRIPTION
**Pull request by zaro0508**
Add two additional public and private subnet to the vpc.

diagram: https://imgur.com/a/FH1Mm

### CODE REVIEW

**Comment by zaro0508** [comment]
We segment 4094 hosts per subnet so theoretically we can setup up to 16 subnets.   There's no reason for 6 subnets except that it's probably all we need now.  We can add subnets later on. 

There's also no reasoning for AZs but I think you're asking why I've put  "PublicSubnetAz0" and "PublicSubnet1Az0" subnets in the same AZ?  I'm not even sure we need 3 public and 3 private subnets.  I was thinking about removing the "PublicSubnetAz0" and "PrivateSubnetAz0"  subnets. 

Let me know your preference on these.

**Comment by xschildw** [comment]
My point was more that we should be able to specify the AZs (i.e. not forced to take 0 and 3 in the list). I agree that 2 public + 2 private should be enough (1 public + 1 private / AZ).

File: cf_templates/vpc.yml [hunk 1]
```diff
@@ -81,6 +88,96 @@ Resources:
...
+  PublicSubnet1Az0:
+    Type: "AWS::EC2::Subnet"
+    Properties:
+      MapPublicIpOnLaunch: true
+      VpcId: !Ref VPC
+      CidrBlock: !Join
+        - '.'
+        - - !Ref VpcSubnetPrefix
+          - !FindInMap [SubnetConfig, Public1, CIDR]
+      AvailabilityZone: !Select
```
Comment (xschildw) [inline cf_templates/vpc.yml]: Can we pass the index as a param?

File: cf_templates/vpc.yml [hunk 2]
```diff
@@ -81,6 +88,96 @@ Resources:
...
+    Type: "AWS::EC2::Subnet"
+    Properties:
+      MapPublicIpOnLaunch: true
+      VpcId: !Ref VPC
+      CidrBlock: !Join
+        - '.'
+        - - !Ref VpcSubnetPrefix
+          - !FindInMap [SubnetConfig, Public2, CIDR]
+      AvailabilityZone: !Select
+        - 3
```
Comment (xschildw) [inline cf_templates/vpc.yml]: Same here

File: cf_templates/vpc.yml [hunk 3]
```diff
@@ -81,6 +88,96 @@ Resources:
...
+  PrivateSubnet1Az0:
+    Type: "AWS::EC2::Subnet"
+    Properties:
+      VpcId: !Ref VPC
+      CidrBlock: !Join
+        - '.'
+        - - !Ref VpcSubnetPrefix
+          - !FindInMap [SubnetConfig, Private1, CIDR]
+      AvailabilityZone: !Select
+        - 0
```
Comment (xschildw) [inline cf_templates/vpc.yml]: Same

File: cf_templates/vpc.yml [hunk 4]
```diff
@@ -81,6 +88,96 @@ Resources:
...
+  PrivateSubnet2Az3:
+    Type: "AWS::EC2::Subnet"
+    Properties:
+      VpcId: !Ref VPC
+      CidrBlock: !Join
+        - '.'
+        - - !Ref VpcSubnetPrefix
+          - !FindInMap [SubnetConfig, Private2, CIDR]
+      AvailabilityZone: !Select
+        - 3
```
Comment (xschildw) [inline cf_templates/vpc.yml]: Same

**Comment by xschildw** [comment]
What is the use for PublicSubnetAz0 and PrivateSubnetAz3? They are setup differently than the other subnets (where public and private are in the same AZ). My request is basically to replace the hardcoded indices 0 and 3 by 2 params 'zone1' and 'zone2' so we can decide in which zones to deploy the subnets.

**Comment by zaro0508** [comment]
Your request is doable.  However we might want to give those parameters more specific names but I guess it would depend on how we want to organize those zones.  for example, do we want to put all public subnets in zone1 and all private subnets in zone2?  then maybe we should call it "publiczone" and "privatezone"?  The alternative is to put each subnet in different zones, in that case we may want to pass in publiczone1, publiczone2, privatezone1, and privatezone2?

**Comment by xschildw** [comment]
Khai, after discussion I understand the change. Please go ahead and pull it in. Thanks!

**Pull request merged**